### ISSUE TITLE
Merging of type hints with phpdoc could be better

### ISSUE THREAD

**Issue by hworld** [bmewburn/intelephense#78]
In this code:

```php
/**
 * @param int[] $mentionIds
 * @param Mention[] $mentions
*/
public static function gatherComments(array $mentionIds, array $mentions)
{
  $mentions[0]->|
}
```

Intelephense thinks that $mentions is just an `array` and not a `Mention[]`. This means it doesn't give any suggestions for the `$mentions[0]->` part. It used to, I believe. Since phpdoc allows you to be more specific on top of the defined type hints in PHP, I think it should try to merge them since it provides the opportunity to have more complex type hints on top of the function definition.

What are your thoughts for this?

**Comment by bmewburn** [comment]
Yes this is a good suggestion. Currently the type declarations override the PHP doc.

**Comment by bmewburn** [comment]
Implemented in 1.0
